katz/src/main/kotlin/katz/free/Yoneda.kt

```diff
@@ -0,0 +1,29 @@
+package katz
+
+typealias YonedaKind<U, A> = HK2<Yoneda.F, U, A>
+
+typealias YonedaF<U> = HK<Yoneda.F, U>
+
+// FIXME using YonedaKind throws a compiler error, but not the expanded form
+interface Yoneda<FU, A> : HK<HK<Yoneda.F, FU>, A> {
```

Comment:
The bug went away when I tried to compile again to get a debug trace. I've reported it anyway to https://youtrack.jetbrains.com/issue/KT-17619